test/tempfile.h

```diff
@@ -55,6 +55,7 @@ class TempFile
 
   ~TempFile()
   {
+    close(fd_);
```

Comment:
I would move this change in the `TempFile` class creation commit.
And then move the remove of `close(tmp_fd)` in the tests when we are moving the test to use `TempFile`. This would remove this commit and avoid to have a "wrong" `TempFile` is our git history.

Reply:
Done